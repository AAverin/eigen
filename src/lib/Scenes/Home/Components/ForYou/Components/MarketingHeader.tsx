import { BorderBox, Box, color, Sans, Separator } from "@artsy/palette"
import { InvertedButton } from "lib/Components/Buttons"
import { Fonts } from "lib/data/fonts"
import React from "react"
import styled from "styled-components/native"

const Container = styled.View`
  padding: 20px;
`

export const MarketingHeader = () => {
  return (
    <Container>
      <BorderBox width="100%" height="220px" background={color("black5")} />

      <Box my={2}>
        <Sans size="3">Introducing a new way to buy on Artsy</Sans>
      </Box>

      {/* TODO: Port Reaction <Button variant='...' /> API over */}
      <InvertedButton
        text="Browse works"
        style={{
          width: 130,
          borderRadius: 2,
          borderColor: "transparent",
        }}
        textStyle={{
          color: "white",
          fontFamily: Fonts.Unica77LLMedium,
          fontSize: 14,
          textShadowColor: "white",
        }}
        key="1"
      />

      <Box mt={4}>
        <Separator />
      </Box>
    </Container>
  )
}
