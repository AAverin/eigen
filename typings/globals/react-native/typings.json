{
  "resolution": "main",
  "tree": {
    "src": "https://raw.githubusercontent.com/DefinitelyTyped/DefinitelyTyped/4c7d03b088d4e71cd9fe9c66ab7bd363049077c3/react-native/react-native.d.ts",
    "raw": "registry:dt/react-native#0.29.0+20160719232534",
    "typings": "https://raw.githubusercontent.com/DefinitelyTyped/DefinitelyTyped/4c7d03b088d4e71cd9fe9c66ab7bd363049077c3/react-native/react-native.d.ts"
  }
}
