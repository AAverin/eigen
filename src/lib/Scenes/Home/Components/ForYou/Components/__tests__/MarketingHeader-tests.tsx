import React from "react"
import "react-native"
import * as renderer from "react-test-renderer"
import { MarketingHeader } from "../MarketingHeader"

describe("getSubjectArtist", () => {
  it("looks correct when rendered", () => {
    const tree = renderer.create(<MarketingHeader />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
