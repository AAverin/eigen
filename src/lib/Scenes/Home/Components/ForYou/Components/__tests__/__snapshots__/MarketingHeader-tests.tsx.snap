// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getSubjectArtist looks correct when rendered 1`] = `
<View
  style={
    Array [
      Object {
        "paddingBottom": 20,
        "paddingLeft": 20,
        "paddingRight": 20,
        "paddingTop": 20,
      },
      undefined,
    ]
  }
>
  <View
    background="#F8F8F8"
    height="220px"
    style={
      Array [
        Object {
          "display": "flex",
          "height": 220,
          "width": "100%",
        },
        Array [
          Object {
            "backgroundColor": "#F8F8F8",
            "borderColor": "#E5E5E5",
            "borderRadius": 2,
            "borderStyle": "solid",
            "borderWidth": 1,
            "height": 220,
            "paddingBottom": 20,
            "paddingLeft": 20,
            "paddingRight": 20,
            "paddingTop": 20,
            "width": "100%",
          },
          undefined,
        ],
      ]
    }
    width="100%"
  />
  <View
    my={2}
    style={
      Array [
        Object {
          "marginBottom": 8,
          "marginTop": 8,
        },
        undefined,
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      className="sc-bdVaJa bDWFJH"
      ellipsizeMode="tail"
      fontFamily="Unica77LL-Regular"
      fontSize={14}
      lineHeight={24}
      style={
        Array [
          Object {
            "fontFamily": "Unica77LL-Regular",
            "fontSize": 14,
            "lineHeight": 24,
          },
          undefined,
        ]
      }
    >
      Introducing a new way to buy on Artsy
    </Text>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    collapsable={undefined}
    hitSlop={undefined}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "rgba(51, 51, 51, 1)",
        "borderColor": "transparent",
        "borderRadius": 2,
        "borderWidth": 1,
        "height": 36,
        "justifyContent": "center",
        "opacity": 1,
        "width": 130,
      }
    }
    testID={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "fontSize": 12,
          },
          Object {
            "fontFamily": "AvantGardeGothicITC",
          },
          Object {
            "color": "rgba(204, 204, 204, 1)",
            "fontFamily": "Unica77LL-Medium",
            "fontSize": 14,
            "opacity": 1,
            "textShadowColor": "white",
          },
        ]
      }
    >
      Browse works
    </Text>
  </View>
  <View
    mt={4}
    style={
      Array [
        Object {
          "marginTop": 32,
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "borderBottomWidth": 0,
            "borderColor": "#E5E5E5",
            "borderStyle": "solid",
            "borderWidth": 1,
            "width": "100%",
          },
          undefined,
        ]
      }
      width="100%"
    />
  </View>
</View>
`;
